driver.go

```diff
@@ -105,3 +105,15 @@ func (d MySQLDriver) OpenConnector(dsn string) (driver.Connector, error) {
 		cfg: cfg,
 	}, nil
 }
+
+// Connect connects to MySQL server.
+// You can use this function to create custom connector which dynamically
+// change the config.  The cfg object must not be changed during calling
+// this function.
+func Connect(ctx context.Context, cfg *Config) (driver.Conn, error) {
```

Comment:
Why use a pointer here if it's sensitive to concurrent changes?

Reply:
Because shallow copy of the config object is not a right way to copy config object.